src/interface/cron/events/addedCollaborator.js

```diff
@@ -26,37 +27,33 @@ agenda.define(jobs.ADDED_COLLABORATOR, async (job, done) => {
       addedCollaborators.length
     );
     await processAddedCollaboratorEvents(addedCollaborators);
-    const lastAddedCollaboratorCheckpoint =
-      getLastAddedCollaboratorCheckpoint({ addedCollaborators, batchSize, latestBlockNumber });
-    if (lastAddedCollaboratorCheckpoint) {
-      await updateAddedCollaboratorCheckpoint(lastAddedCollaboratorCheckpoint);
-    }
     done();
   } catch (err) {
     console.error("Error in job", jobs.ADDED_COLLABORATOR, err);
     done(err);
   } finally {
+    if (addedCollaborators.length > 0) {
+      await updateAddedCollaboratorCheckpoint(addedCollaborators[addedCollaborators.length - 1].blockNumber);
+    }
     console.log("Job done", jobs.ADDED_COLLABORATOR);
   }
 });
 
-async function getLatestBlockNumberFromSubgraph() {
-  const response = await axios.get(STATUS_API_URL);
-  const statusObject = response?.data?.data['indexingStatusForCurrentVersion'] || {};
-  const chains = statusObject.chains || [];
-  const firstObject = chains.pop();
-  return parseInt(firstObject?.latestBlock?.number, 10) || 0;
-}
-
 async function fetchAddedCollaboratorCheckpoint() {
   const eventProcessed = await EventProcessor.findOne({});
   return eventProcessed ? eventProcessed.addedCollaborator : 0;
 }
 
 async function fetchAddedCollaboratorEvents(checkpoint, itemCount) {
+  const fetchedEvents = await fetchAddedEventsID(EVENT_NAME);
   const response = await axios.post(API_URL, {
     query: `{
-      ${EVENT_NAME}(first: ${itemCount || 5}, orderDirection: asc, orderBy: blockNumber, where: { blockNumber_gte : ${checkpoint} }) {
+      ${EVENT_NAME}(first: ${itemCount || 5}, orderDirection: asc, orderBy: blockNumber, 
+        where: {
+          blockNumber_gte : ${checkpoint},
+          id_not_in:[${fetchedEvents.map(event => `"${event}"`).join(', ')}]

```

Comment:
This should also work `id_not_in: fetchedEvents`?

Reply:
nope. Got error with that. 
query needs the list in exact list format. 
`id_not_in: [ "abc, "bcd" ]`

Reply:
Can you paste an example fetchedEvents array here .. it should be in this format only right

Reply:
![image](https://github.com/fileverse/fileverse-coordinator/assets/26124625/7d7ec6c3-e6c8-480a-a83c-43dfaf31c5fe)